sonic-sensormond/scripts/sensormond

```diff
@@ -0,0 +1,561 @@
+                low_critical_threshold = try_get(vsensor.get_low_critical_threshold)
+
+            warning = False
+            if voltage != NOT_AVAILABLE and voltage_status.set_over_threshold(voltage, high_threshold):
+                self._log_on_status_changed(not voltage_status.over_threshold,
+                                            'High voltage warning cleared: {} voltage restored to {}mV, high threshold {}mV'.
+                                            format(name, voltage, high_threshold),
+                                            'High voltage warning: {} current voltage {}mV, high threshold {}mV'.
+                                            format(name, voltage, high_threshold)
+                                            )
+            warning = warning | voltage_status.over_threshold
+
+            if voltage != NOT_AVAILABLE and voltage_status.set_under_threshold(voltage, low_threshold):
+                self._log_on_status_changed(not voltage_status.under_threshold,
+                                            'Low voltage warning cleared: {} voltage restored to {}mV, low threshold {}mV'.
+                                            format(name, voltage, low_threshold),
+                                            'Low voltage warning: {} current voltage {}mV, low threshold {}mV'.
+                                            format(name, voltage, low_threshold)
+                                            )
+            warning = warning | voltage_status.under_threshold
+
+            fvs = swsscommon.FieldValuePairs(
+                [('voltage', str(voltage)),
+                ('minimum_voltage', str(minimum_voltage)),
+                ('maximum_voltage', str(maximum_voltage)),
+                ('high_threshold', str(high_threshold)),
+                ('low_threshold', str(low_threshold)),
+                ('warning_status', str(warning)),
+                ('critical_high_threshold', str(high_critical_threshold)),
+                ('critical_low_threshold', str(low_critical_threshold)),
+                ('is_replaceable', str(is_replaceable)),
+                ('timestamp', datetime.now().strftime('%Y%m%d %H:%M:%S'))
+                ])
+
+            self.table.set(name, fvs)
+            if self.is_chassis_system and self.chassis_table is not None:
+                self.chassis_table.set(name, fvs)
+        except Exception as e:
+            self.log_warning('Failed to update vsensor status for {} - {}'.format(name, repr(e)))
+
+    def _remove_vsensor_from_db(self, vsensor, parent_name, vsensor_index):
+        name = try_get(vsensor.get_name, '{} vsensor {}'.format(parent_name, vsensor_index + 1))
+        self.table._del(name)
+
+        if self.chassis_table is not None:
+            self.chassis_table._del(name)
+    
+#
+# currentUpdater  ======================================================================
+#
+class CurrentUpdater(logger.Logger):
```

Comment:
Can we create a base SensorUpdater class that these can inherit? The code looks nearly identical, except for some variable names and strings in logs (which could be handled through some object variable saying what type of sensor it is).

Reply:
i had thought about it. However it seemed like it would make the code have bunch of if/else and if we add thermal sensors/altitude sensors in the mix (later) it would make it more complex.